### DISCUSSION TITLE
useController vs <Controller />

### DISCUSSION THREAD

**Discussion by krau5** [react-hook-form/react-hook-form#6157] [General] [answered]
Can someone explain me what's better to use: useController or <Controller />?
Which advantages or disadvantages can u highlight?

**Comment by bluebill1049** [comment, marked as answer]
<Controller /> is when you want to quickly implement a component and used a wrapper

`useController` is when you want to build a reusable component. (I use this at work, it's cleaner without the Controller as wrapper)

**Reply by krau5** [reply]
one mo question, how can I get the same result with useController as in the link below using <Controller />
https://react-hook-form.com/advanced-usage#TransformandParse

**Reply by bluebill1049** [reply]
it's the same approach, the only difference is one is render prop, and the other is at the component level with hook api.

**Reply by vltansky** [reply]
I created this example to demonstrate the difference. Hope it helps someone :)
https://stackblitz.com/edit/stackblitz-starters-gyqpdm?file=app%2Fpage.tsx

**Reply by krau5** [reply]
@vltansky 3 years have passed 😄 Anyway, it looks awesome!

**Reply by vltansky** [reply]
I searched for example of difference and didn't find one, and this discussion was among top results on google, hopefully it will save someone time :)
